tests/chat/test_chat.py

```diff
@@ -1,54 +1,75 @@
+import os
 import tempfile
 import time
+import mongomock
 from channels.testing import ChannelsLiveServerTestCase
+from mongoengine import connect, disconnect
 from selenium import webdriver
 from selenium.webdriver.chrome.options import Options
 from selenium.webdriver.chrome.service import Service
 from selenium.webdriver.common.action_chains import ActionChains
 from selenium.webdriver.common.keys import Keys
 from webdriver_manager.chrome import ChromeDriverManager
+from users.documents import UserDocument, UserRoleDocument, UserRoleEnum
 
+os.environ['DJANGO_SETTINGS_MODULE'] = 'core.test_settings'
 
-class ChatTests(ChannelsLiveServerTestCase):
-    """
-    Integration tests for Django Channels chat using Selenium.
 
-    Tests ensure:
-    - Messages are received by all clients in the same room.
-    - Messages are not received by clients in other rooms.
-    """
+class ChatTests(ChannelsLiveServerTestCase):
     serve_static = True
 
     @classmethod
     def setUpClass(cls):
+        """
+        Set up in-memory MongoDB, test user, and Selenium WebDriver before running tests.
+        """
         super().setUpClass()
+        disconnect()
+        connect(
+            db='mongoenginetest',
+            host='mongodb://localhost',
+            mongo_client_class=mongomock.MongoClient
+        )
+        role = UserRoleDocument.objects(role=UserRoleEnum.USER).first()
+        if not role:
+            role = UserRoleDocument(role=UserRoleEnum.USER)
+            role.save()
+
+        cls.user = UserDocument(
+            email="testuser@example.com",
+            first_name="Test",
+            last_name="User",
+            password="testpass",
+            role=role
+        )
+        cls.user.save()
```

Comment:
The password is hardcoded and not hashed. If your UserDocument requires a hashed password, ensure this is done. Add a comment warning this is for test only.
Add a check or exception handling for duplicate users.
Add a teardown process to remove test users after tests run to avoid polluting the test DB.